.github/workflows/integration-test.yml

```diff
@@ -2,113 +2,84 @@ name: Integration Tests
+        with:
+          opensearch-version: ${{ env.OPENSEARCH_VERSION }}
+          plugin-name: ${{ env.PLUGIN_NAME }}
+          setup-script-name: setup
 
-      # - name: Get OpenSearch Dashboards version
-      #   id: osd_version
-      #   run: |
-      #     echo "::set-output name=osd_version::$(jq -r '.opensearchDashboards.version' ./OpenSearch-Dashboards/plugins/security-dashboards-plugin/package.json)"
-      
-      # - name: Check OpenSearch Dashboards release tag
-      #   run: |
-      #     cd ./OpenSearch-Dashboards
-      #     git checkout tags/${{ steps.osd_version.outputs.osd_version }} -b v${{ steps.osd_version.outputs.osd_version }}
-      
-      - name: Get node and yarn versions
-        id: versions
-        run: |
-          echo "::set-output name=node_version::$(cat ./OpenSearch-Dashboards/.node-version)"
-          echo "::set-output name=yarn_version::$(jq -r '.engines.yarn' ./OpenSearch-Dashboards/package.json)"
-      
-      - name: Setup node
-        uses: actions/setup-node@v1
+      - id: install-dashboards
+        uses: ./.github/actions/install-dashboards
         with:
-          node-version: ${{ steps.versions.outputs.node_version }}
-          registry-url: 'https://registry.npmjs.org'
-      
-      - name: Install correct yarn version for OpenSearch Dashboards
-        run: |
-          npm uninstall -g yarn
-          echo "Installing yarn ${{ steps.versions_step.outputs.yarn_version }}"
-          npm i -g yarn@${{ steps.versions.outputs.yarn_version }}
-      
-      - name: Check OpenSearch Running
-        continue-on-error: true
-        run: curl -XGET https://localhost:9200 -u 'admin:admin' -k
+          plugin_name: security-dashboards-plugin
 
-      - name: Bootstrap OpenSearch Dashboards
-        run: |
-          cd ./OpenSearch-Dashboards
-          yarn osd bootstrap
-          node scripts/build_opensearch_dashboards_platform_plugins.js
-      
-      - name: Run integration tests
+      - name: Start Dashboards in background
+        run: node scripts/build_opensearch_dashboards_platform_plugins.js
+        working-directory: ${{ steps.install-dashboards.outputs.dashboards-directory }}
+
+      - name: Run integration tests on Linux
+        if: ${{ runner.os == 'Linux' }}
         run: |
           echo "check if opensearch is ready"
           curl -XGET https://localhost:9200 -u 'admin:admin' -k
-          cd ./OpenSearch-Dashboards/plugins/security-dashboards-plugin
           yarn test:jest_server --coverage
+        working-directory: ${{ steps.install-dashboards.outputs.plugin-directory }}
+
+      - name: Run integration tests on Windows
+        if: ${{ runner.os == 'Windows' }}
+        run: |
+          echo "check if opensearch is ready"
+          curl -XGET https://localhost:9200 -u 'admin:admin' -k
+          node .\test\run_jest_tests.js --config .\test\jest.config.server.js --testPathIgnorePatterns saml_auth.test.ts
```

Comment:
For my understanding, once SAML auth tests are fixed, `--testPathIgnorePatterns` will be removed, correct?

Reply:
Yes, same as `main` and `2.x`, we will remove that once we figure out a fix.